cmd/snap-update-ns/change.go

```diff
@@ -100,12 +100,12 @@ func (c *Change) createPath(path string, pokeHoles bool, sec *Secure) ([]*Change
 	case "symlink":
 		err = sec.MksymlinkAll(path, mode, uid, gid, c.Entry.XSnapdSymlink())
 	}
-	if err2, ok := err.(*ReadOnlyFsError); ok && pokeHoles {
+	if err2, ok := err.(MimicRequiredError); ok && pokeHoles {
```

Comment:
I'm not sure about the interface, how about this:
```
func mimicRequired(err error) (needsMimic bool, path string) {
    switch err.(type) {
        case *ReadOnlyFsError:
            roErr := err.(*ReadOnlyFsError)
            return true, roErr.Path
    }
    return false, ""
}
```

Then in the check:
```
if needsMimic, path := mimicRequired(err); needsMimic && pokeHoles {
    ...
}
```

Reply:
Hmm, I could do that, there are just two types of errors really. One is the current ReadOnlyFsError and the other one is the upcoming TrespassingError. I could do a type assertion if you think this is cleaner.